tracing-core/src/callsite.rs

```diff
@@ -125,24 +142,25 @@ pub struct Identifier(
 /// [`Interest::sometimes()`]: ../subscriber/struct.Interest.html#method.sometimes
 /// [`Subscriber`]: ../subscriber/trait.Subscriber.html
 pub fn rebuild_interest_cache() {
-    let mut registry = REGISTRY.lock().unwrap();
-    registry.rebuild_interest();
+    REGISTRY.rebuild_interest();
 }
 
 /// Register a new `Callsite` with the global registry.
 ///
 /// This should be called once per callsite after the callsite has been
 /// constructed.
 pub fn register(callsite: &'static dyn Callsite) {
-    let mut registry = REGISTRY.lock().unwrap();
-    registry.rebuild_callsite_interest(callsite);
-    registry.callsites.push(callsite);
+    REGISTRY.rebuild_callsite_interest(callsite);
+    REGISTRY.callsites.push(callsite);
 }
 
 pub(crate) fn register_dispatch(dispatch: &Dispatch) {
-    let mut registry = REGISTRY.lock().unwrap();
-    registry.dispatchers.push(dispatch.registrar());
-    registry.rebuild_interest();
+    REGISTRY
+        .dispatchers
+        .lock()
+        .unwrap()
+        .push(dispatch.registrar());
+    REGISTRY.rebuild_interest();
```

Comment:
this will lock the registry twice (once in rebuild_interest, and once when we lock the dispatchers). it might be better nice if we used one `MutexGuard` (and thus one `lock()` call) for both operations.